ipatests/test_integration/test_ipa_cert_fix.py

```diff
@@ -260,3 +261,68 @@ def test_renew_expired_cert_with_kra(self, expire_cert_critical):
         self.master.run_command(['ipa-cert-fix', '-v'], stdin_text='yes\n')
 
         check_status(self.master, 12, "MONITORING")
+
+
+class TestCertFixReplica(IntegrationTest):
+
+    num_replicas = 1
+
+    @classmethod
+    def install(cls, mh):
+        tasks.install_master(
+            mh.master, setup_dns=False, extra_args=['--no-ntp']
+        )
+        tasks.install_replica(
+            mh.master, mh.replicas[0],
+            setup_dns=False, extra_args=['--no-ntp']
+        )
+
+    def test_renew_expired_cert_replica(self):
+        """Test renewal of certificates on replica with ipa-cert-fix
+
+        This is to check that ipa-cert-fix renews the certificates
+        on replica
+
+        related: https://pagure.io/freeipa/issue/7885
+        """
+        move_date(self.master, 'stop', '+3years+1days')
+
+        # wait for cert expiry
+        check_status(self.master, 8, "CA_UNREACHABLE")
+
+        self.master.run_command(['ipa-cert-fix', '-v'], stdin_text='yes\n')
+
+        check_status(self.master, 9, "MONITORING")
+
+        # move system date to expire cert on replica
+        move_date(self.replicas[0], 'stop', '+3years+1days')
```

Comment:
Are you keeping the replica at a different time so that replication of the updated certs doesn't work so that you can test the state of it in a more controlled way?

Reply:
I didn't get that. But I moved the date with exact time frame on master too.
So now, master have updated certs. Replica have same date as master and 3 certs(httpd/ldap/pkinit) got renewed(replicated from master) and remaining needs to be resubmit and RA agent will be fixed by `ipa-cert-fix`